app/src/main/java/com/poupa/vinylmusicplayer/provider/BlacklistStore.java

```diff
@@ -31,7 +31,7 @@ public BlacklistStore(final Context context) {
     @Override
     public void onCreate(@NonNull final SQLiteDatabase db) {
         db.execSQL("CREATE TABLE IF NOT EXISTS " + BlacklistStoreColumns.NAME + " ("
-                + BlacklistStoreColumns.PATH + " STRING NOT NULL);");
+                + BlacklistStoreColumns.PATH + " TEXT NOT NULL);");
```

Comment:
I have a doubt here: Does this requires DB migration, and storage version bump?

Reply:
Probably so. I'm bad at explaining things, so I prefer to just let the lints do the explaining.

![image](https://user-images.githubusercontent.com/32376686/89819977-3f8c1880-db1a-11ea-883d-6fc9675e7fbd.png)

Reply:
My comment is not specifically against the `string` to `text` transition, I was asking if this change (if makes sense, which looks to be the case) would require a database upgrade (i.e. converting existing data to the new column type) and if yes, it should:
- require a DB version bump (i.e. letting app binary version N that opens database created by version M knows that it is dealing with a probably incompatible DB layout.
- optionally having upgrade/downgrade SQL scripts to handle the transition smoothly for user.

Given the fact that we are dealing with FS paths here, unwanted data loss (as explained by lint) is unlikely imho.

Reply:
Ah, ok. I don't know whether this will require a DB upgrade. I'm also not very familiar with SQLite.